agent/engine/docker_task_engine.go

```diff
@@ -597,6 +612,20 @@ func (engine *DockerTaskEngine) createContainer(task *api.Task, container *api.C
 	seelog.Infof("Created container name mapping for task %s - %s -> %s", task, container, containerName)
 	engine.saver.ForceSave()
 
+	// Create metadata directory and file then populate it with common metadata of all containers of this task
+	// Afterwards add this directory to the container's mounts if file creation was successful
+	// CreateMetadata has a call to docker API for the client version so this may be block our container creation
+	// But we require the host Config binds to be properly set before we create the container
+	if engine.cfg.ContainerMetadataEnabled {
+		newBinds, mderr := engine.metadataManager.CreateMetadata(hostConfig.Binds, task, container)
+		if mderr != nil {
+			seelog.Errorf("Failed to create metadata file for container %s of task %s: %v", container, task, mderr)
+		} else {
+			// Update hostConfig binds if metadata creation is successful
+			seelog.Debugf("Created metadata file for container %s of task %s", container, task)
+			hostConfig.Binds = newBinds
```

Comment:
This adds the metadata directory to the mounted volumes of the container, which is necessary to expose the file to the container. I think I should comment this section better.